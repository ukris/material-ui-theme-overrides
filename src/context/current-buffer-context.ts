import { createContext } from 'react'
import { Buffer } from 'types'

interface ICurrentBufferContextProps {
    currentBuffer?: Buffer[]
}

const CurrentBufferContext  = createContext({} as ICurrentBufferContextProps)

export default CurrentBufferContext