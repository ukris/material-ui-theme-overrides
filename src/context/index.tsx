import CurrentBufferContext from './current-buffer-context'
import TableContext from './table-context'
import GlobalContext from './global-context'
import ProjectContext from './project-context'

export { CurrentBufferContext,  GlobalContext, ProjectContext, TableContext }