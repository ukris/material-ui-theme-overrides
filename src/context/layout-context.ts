import { createContext } from 'react'

interface ILayoutContext {
    fullScreen?: boolean
    collapsed?: boolean
    miniMenu?: boolean
    setFullScreen: (full:boolean) => void
    setCollapsed: (collapsed: boolean) => void
    setMiniMenu: (mini:boolean) => void
    width: number
    height: number
    setWidth: (width: number) => void
    setHeight: (height: number) => void
};

const LayoutContext = createContext({} as ILayoutContext);

export default LayoutContext;