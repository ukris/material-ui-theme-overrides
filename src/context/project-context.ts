import { createContext } from 'react'

interface IProjectContextProps {
    currentProject: any,
    setCurrentProject: (id:any) => void
    projects?: any[],
    setProjects: (project:any) => void
};

const ProjectContext = createContext({} as IProjectContextProps);

export default ProjectContext;