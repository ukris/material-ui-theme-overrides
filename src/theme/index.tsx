import { themePalette, LIGHT, DARK } from './palette'
export type LIGHTORDARK = 'light' | 'dark'
export { default } from './theme-provider'
export { default as theme } from './theme'
export { default as makePalette,  } from './palette'
export { default as typography } from './typography'
export { default as ThemeContext } from './theme-context'
export { default as ThemeProvider } from './theme-provider'
export { themePalette, LIGHT, DARK }

const Themes = {
  basil: null,
  crane: null,
  material: null,
  materialDark: null,
  pinky: null,
  rally: null,
  reply: null,
}
export type Color = 'primary' | 'success' | 'info' | 'accent' | 'warning' | 'danger' | 'error' | 'default'
export type ColorValue = {
    light: string,
    main: string,
    dark: string,
    contrastText: string
}
export type ColorObj = {
  [key in Color] : ColorValue
}

export type ThemeType = keyof typeof Themes
export type ThemeZoom = {
  theme: ThemeType
  zoom: number
}
export const AVAILABLE_THEMES = Object.keys(Themes) as ThemeType[]
export const unit = 'rem'
export const tablePadding = [1,0]
export const FONT_BASIS = 2