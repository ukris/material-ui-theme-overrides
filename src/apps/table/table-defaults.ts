import { capitalize } from 'helpers/string'
import { CellType, Column, CellTypeArray } from './types'
import { renderDateRange, renderLocation, renderLabel, renderRating, renderPercentage, renderTags, renderISOWithFlag, renderImage, renderPeople, renderEllipsis } from './cell-renderers'
import { editDateRange, editAddress,  editISOWithFlag, editLabel, editPeople, editEllipsis, editTags } from './cell-editors'
import  { FONT_BASIS } from 'theme'
export const cellTypeUserDefaults = {
    default:{
        align: 'left',
        width: 5
    },
    [CellType.dateRange]: {
        width: 10,
        renderer: renderDateRange,
        editor: editDateRange
    },
    [CellType.location]: {
        width: 15,
        renderer: renderLocation,
        editor: editAddress
    },
    [CellType.rating]: {
        width: 8,
        renderer: renderRating,
        number: true,
        step: 0.5,
        min: 1,
        max: 5
    },
    [CellType.progressPercentage]:{
        width: 15,
        renderer: renderPercentage,
        number: true,
        min: 0,
        max: 100
    },
    [CellType.label]:{
        width: 7,
        renderer: renderLabel,
        editor: editLabel
    },
    [CellType.tags]:{
        width: 16,
        renderer: renderTags,
        editor: editTags
    },
    [CellType.flagWithISO]:{
        renderer: renderISOWithFlag,
        editor: editISOWithFlag,
        width: 6
    },
    [CellType.image]:{
        renderer: renderImage
    },
    [CellType.people]:{
        renderer: renderPeople,
        editor: editPeople,
        width: 9
    },
    [CellType.ellipsis]:{
        renderer: renderEllipsis,
        editor: editEllipsis,
        width: 10
    }
  }

  export const getColumnDefinitionWithDefaults = (col: Column) => {
    const { type = CellType.text } = col
    const nm = type as string

    // @ts-ignore
    const typeDefs = cellTypeUserDefaults[type] || {} 
    const def= {...cellTypeUserDefaults.default,
                ...typeDefs }
    const ret: Column = {...def, ...col}
    if (!ret.measure) {
        const measure = ~nm.indexOf('range') 
                    ? true 
                    : ~nm.indexOf('date') 
                    ? true
                    : ~nm.indexOf('number') 
                    ? true 
                    : ~nm.indexOf('amount') 
                    ? true
                    : ~nm.indexOf('percent') 
                    ? true : false
        if (measure) {
            ret.measure = measure
        }
    }
    if (!ret.label) {
      ret.label = capitalize(ret.id)
    }
    ret.width = ret.width * FONT_BASIS/2
    return ret
}