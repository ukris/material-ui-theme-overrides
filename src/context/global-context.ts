import { ThemeZoom }     from 'theme'
import { Buffer, FontSize } from 'types'
import  defaults, { Settings } from '../settings'
import { Cell, CellValue, Column } from 'apps/table/types'

const FULL_SCREEN = 'full_screen'
const THEME_ZOOM = 'theme_zoom'
const SETTINGS = 'settings'
const CURRENT_BUFFER = 'buffer'
const BUFFERS = 'buffers'
class LocalStorageMock {
    store:any={}
    constructor() {
      this.store = {};
    }
  
    clear() {
      this.store = {};
    }
  
    getItem(key:string) {
      return this.store[key] || null;
    }
  
    setItem(key:string, value:string) {
      this.store[key] = value.toString();
    }
  
    removeItem(key:string) {
      delete this.store[key];
    }
  };
  if (!global.localStorage) {
      // @ts-ignore
      global.localStorage = new LocalStorageMock()
  }
interface IGlobalContext {
    buffers: Buffer[]
    nextDomainId: number
    nextProjectId: number
    nextBoardId: number
    nextLaneId: number
    settings: Settings
    fullScreen: boolean
    palette?: any
    themeZoom: ThemeZoom,
    fontSize: FontSize,
    setThemeZoom: (themeZoom: ThemeZoom) => void
    setSettings : (settings: Settings) => void
    setFullScreen: (isFullScreen: boolean) => void
    focusedCell?: Cell | null
    currentBuffer?: Buffer
    domainId?: number
    projectId?: number
    boardId?: number
    laneBy?: number
    laneId?: string
    rows: any[]
    columns: Column[]
    anchorEl?: any
    setCurrentBuffer: (buffers: Buffer) => void
    setBuffers: (buffers: Buffer[]) => void
    setDomainId:(domainId: number) => void
    setProjectId:(projectId: number) => void
    setBoardId:(boardId: number) => void
    setLaneId:(laneId: string) => void
    setLaneBy:(labeBy: number) => void
    setRows: (rows:any[]) => void
    setColumns: (columns: Column[]) => void
    getColumn: (idx: number) => Column
    updateCell: (cell: Cell, newVal: CellValue) => void
    setFocusedCell:  (cell: Cell|null) =>void
    setAnchorEl: (el:any) => void
    theme: any
    setTheme: (theme:any)  => void
    /*

    setDomainId: (domainId: number) => void
    setProjectId: (projectId: number) => void
    setBoardId: (boardId: number) => void
    setLaneBy?: (laneBy?: string) => void // split by/ group by colId
    setLaneId: (laneId: string) => void // example  laneby status, laneId = 'FINISHED, 'INPROGRESS' etc 
    setRowIndex: (idx: number) => void
    setColIndex: (idx: number) => void
    setRowId: (id: string) => void
    setColId: (id: string) => void
    setRange: (range: number[]) => void
    */
}

const fullScreen = !!(localStorage && localStorage.getItem(FULL_SCREEN))
const tZoom = localStorage && localStorage.getItem(THEME_ZOOM)
const themeZoom:ThemeZoom = tZoom ? JSON.parse(tZoom) as ThemeZoom : { zoom: 1, theme: 'material'}
const defaultS = localStorage && localStorage.getItem(SETTINGS)
const defaultSettings: Settings = defaultS ? JSON.parse(defaultS) : defaults
let lBuffer = localStorage && localStorage.getItem(CURRENT_BUFFER)
const currentBuffer:Buffer = lBuffer ? JSON.parse(lBuffer) : []
lBuffer = localStorage && localStorage.getItem(BUFFERS)
const buffers:Buffer[] = lBuffer ? JSON.parse(lBuffer) : []

export const GlobalContext: IGlobalContext = {
    buffers,
    currentBuffer,
    nextDomainId: 2,
    nextProjectId: 2,
    nextBoardId: 2,
    nextLaneId: 2,
    fullScreen,
    themeZoom,
    settings: defaultSettings,
    theme: null,
    fontSize: {
        xs: '0.5rem',
        sm: '0.75rem',
        md: '1rem',
        lg: '1.25rem',
        xl: '1.5rem'
    },
    columns: [],
    rows: [],
    setFocusedCell: (cell: Cell|null)  => {
        GlobalContext.focusedCell = cell;
    },
    setThemeZoom: (themeZoom:ThemeZoom) => {
        (async() => localStorage && localStorage.setItem(THEME_ZOOM, 
            JSON.stringify(themeZoom)))()
        GlobalContext.themeZoom = themeZoom
    },
    setSettings: (settings: Settings) => {
        (async() => localStorage  && localStorage.setItem(SETTINGS, 
            JSON.stringify(settings)))()
        GlobalContext.settings = settings
    },
    setCurrentBuffer: (buffer: Buffer) => {
        (async() => localStorage  && localStorage.setItem(CURRENT_BUFFER, 
            JSON.stringify(buffer)))()
        GlobalContext.currentBuffer = buffer
    },
    setBuffers: (buffers: Buffer[]) => {
        (async() => localStorage  && localStorage.setItem(BUFFERS, 
            JSON.stringify(buffers)))()
        GlobalContext.buffers = buffers
    },
    setFullScreen: (screen: boolean) => {
        (async() => localStorage  && localStorage.setItem(FULL_SCREEN, screen.toString()))()
        GlobalContext.fullScreen = screen
    },
    setDomainId: (domainId: number) => {
        GlobalContext.domainId = domainId
    },
    setProjectId: (projectId: number) => {
        GlobalContext.projectId = projectId
    },
    setBoardId: (boardId: number) => {
        GlobalContext.boardId = boardId
    },
    setLaneId:(laneId: string) => {
        GlobalContext.laneId = laneId
    },
    setLaneBy:(laneBy: number) => {
        GlobalContext.laneBy = laneBy
    },
    setRows: (rows: any[]) => {
        GlobalContext.rows = rows
    },
    setColumns: (columns: Column[]) => {
        GlobalContext.columns = columns
    },
    getColumn: (idx:number) => GlobalContext.columns[idx],
    updateCell: (cell: Cell, newVal: CellValue) => {
        const { rowIndex, colIndex } = cell
        GlobalContext.rows[rowIndex][colIndex] = newVal
    },
    setAnchorEl: (el:any) => {
        GlobalContext.anchorEl = el
    },
    setTheme: (theme: any) => {
        GlobalContext.theme = theme
    }
}

export default GlobalContext