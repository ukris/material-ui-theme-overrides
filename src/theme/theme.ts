import { createMuiTheme, responsiveFontSizes} from '@material-ui/core'
import PaletteOptions from '@material-ui/core/styles/createMuiTheme';
import { GlobalContext } from 'context/global-context'
import { FONT_BASIS, makePalette, ThemeZoom, unit, tablePadding } from './index'
import { roundNum } from 'helpers'
// https://www.sipios.com/blog-tech/how-to-use-styled-components-with-material-ui-in-a-react-app
import typography from './typography'

function createTheme(variant: ThemeZoom):any {
  let palette:any = makePalette(variant)
  const { zoom } = variant
  const { colors } = palette
  delete palette.colors
  GlobalContext.palette = colors
  GlobalContext.palette.type = palette.type
  GlobalContext.palette.contrastText = palette.contrastText
  palette = {...palette, ...colors}
  GlobalContext.setThemeZoom(variant)
  const theme = createMuiTheme({
    zoomFontSize: (factor:number):string => `${roundNum(0.30 * factor * zoom)}${unit}`,
    zoomSpacing: (factor: number) => spacing(roundNum(factor * zoom)),
    fontSize: GlobalContext.fontSize,
    spacing: factor => `${roundNum(0.60 * factor)}${unit}`,
    lineHeight: GlobalContext.fontSize
  });
  const { spacing, zoomSpacing, zoomFontSize } = theme
  const fontSize =  {
      xs: zoomFontSize(FONT_BASIS),
      sm: zoomFontSize(FONT_BASIS+1),
      md: zoomFontSize(FONT_BASIS+2),
      lg: zoomFontSize(FONT_BASIS+3),
      xl: zoomFontSize(FONT_BASIS+4)
  }
  const lineHeight =  {
    xs: zoomFontSize(FONT_BASIS+0.5),
    sm: zoomFontSize(FONT_BASIS+1.5),
    md: zoomFontSize(FONT_BASIS+2.5),
    lg: zoomFontSize(FONT_BASIS+3.5),
    xl: zoomFontSize(FONT_BASIS+4.5)
  }
  let padding:string = tablePadding[0].toString()
  if (tablePadding[0]) {
    padding+= unit
  }
  padding+= ' '
  padding+= tablePadding[1].toString()
  if (tablePadding[1]) {
    padding+= unit
  }
  GlobalContext.fontSize = fontSize
  const modTheme:any = createMuiTheme(
    /**
     * @see https://material-ui.com/customization/themes/#theme-configuration-variables
     * https://material-ui.com/customization/typography/
     */
    {
      palette,
      zoomFontSize,
      typography,
      spacing,
      zoomSpacing,
      fontSize,
      lineHeight,
      props: {
        MuiButtonBase: {
          disableRipple: true
        },
        MuiButton: {
            variant: 'contained',
            disableElevation: true,
        },
        MuiTextField: {
            variant: 'outlined'
        }
      },
      zIndex: {
        appBar: 1200,
        drawer: 1100,
      },
      overrides: {
        MuiDrawer: {
          paper: {
            background: '#18202c',
          },
        },
        MuiPaper: {
          // Name of the component ⚛️ / style sheet
          elevation0: {
            // Name of the rule
            boxShadow: '0 0 14px 0 rgba(53,64,82,.05)', // Some CSS
          },
        },
        MuiTableCell: {
          root: { 
            borderBottom: '1px solid rgba(224, 224, 224, .5)', 
            lineHeight: lineHeight.md,
            fontSize: fontSize.md,
            backgroundColor: theme.palette.background.default,
          },
        },
        MuiAutocomplete: {
          root: {
            backgroundColor: theme.palette.background.default,
            color: theme.palette.getContrastText(theme.palette.background.default)
          }
        },
        MuiButton: {
          root: {
            color: theme.palette.primary.contrastText,
          }
        },
        MuiTextField: {
          root: {
            variant: 'outlined'
          }
        }
    }}
  )
  //  return responsiveFontSizes(modTheme)
  GlobalContext.theme = theme
  return modTheme
}

export default createTheme