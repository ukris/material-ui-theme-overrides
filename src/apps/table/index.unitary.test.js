import React from 'react'
import renderer from 'react-test-renderer'
import { mount, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Table from './index'
import { GlobalContext, TableContext } from 'context'
import { ThemeContext, ThemeProvider } from 'theme'
import { getColumnDefinitionWithDefaults } from './table-defaults'
import backendMockdata from 'mocks/table.json'

configure({ adapter: new Adapter() })

const tableData = {
    ...backendMockdata,
    columns: backendMockdata.columns.map( column => getColumnDefinitionWithDefaults(column))
}

const themeZoom = GlobalContext.themeZoom;
const setThemeZoom = () => true

describe('Table Component | One row', () => {
    const component = (
        <TableContext.Provider
        value={{
          editable: false,
          setEditable: () => false,
          focusedCell: undefined,
          setFocusedCell: () => false
        }}>
        <ThemeContext.Provider value={{ themeZoom, setThemeZoom }}>
            <ThemeProvider>
                <Table columns={tableData.columns} rowsData={tableData.rows} editable={tableData.editable}/>
            </ThemeProvider>
        </ThemeContext.Provider>
      </TableContext.Provider>
    )
    const wrapper = mount(component)
    it('renders component correctly', () => {
        expect(wrapper.find('[data-test="table-container"]').length).toEqual(1)
    });

    it('matches snapshot', () => {
        const tree = renderer.create(component,).toJSON()

        expect(tree).toMatchSnapshot()
    });

    // it('rendered ellipsis component', () => {
    //     expect(wrapper.find('.table-ellipsis-container').length).toEqual(1)
    // });

    // it('rendered progressbar component', () => {
    //     expect(wrapper.find('[data-test="table-progressbar"]').length).toEqual(1)
    // });
    
    // it('rendered tags component', () => {
    //     expect(wrapper.find('.MuiBox-root[data-test="table-tags"]').length).toEqual(1)
    // });
    
    // it('rendered peopledropdown component', () => {
    //     expect(wrapper.find('[data-test="table-peopledropdownselect"]').length).toEqual(1)
    // });
    
    // it('rendered daterangepicker component', () => {
    //     expect(wrapper.find('[data-test="table-daterangepicker"]').length).not.toEqual(0)
    //     expect(wrapper.find('[data-test="table-daterangepicker"]').find('input').length).toEqual(1)
    // });

    // it('rendered star rating component', () => {
    //     expect(wrapper.find('.MuiRating-root').length).toEqual(1)
    // });
    
    // it('rendered address component', () => {
    //     expect(wrapper.find('[data-test="table-address"]').length).toEqual(1)
    // });

    // it('rendered country component', () => {
    //     expect(wrapper.find('[data-test="table-country"]').length).toEqual(1)
    // });
})
