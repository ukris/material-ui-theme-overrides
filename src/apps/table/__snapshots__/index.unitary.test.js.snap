// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component | One row matches snapshot 1`] = `" "`;
